### PR TITLE
Remove GUI_TESTING code entirely

### PR DESCRIPTION
**Pull request by fmoc**
The  `GUI_TESTING` code doesn't seem to be used anyway, but is causing issues with the changed compilation flags. Therefore, it makes sense to just remove it from the code base.

### CODE REVIEW

File: src/gui/socketapi/socketapi.cpp [hunk 1]
```diff
@@ -139,7 +139,7 @@ QObject *findWidget(const QString &queryString, const QList<QWidget *> &widgets
     } else {
         QList<QObject *> matches;
         std::copy_if(objects.constBegin(), objects.constEnd(), std::back_inserter(matches), [&](QObject *widget) {
-            return widget->inherits(queryString.toLatin1());
+            return widget->inherits(queryString.toLatin1().constData());
```
Comment (TheOneRing) [inline src/gui/socketapi/socketapi.cpp:142]: hmm utf8 instead of latin1?
not sure

File: src/gui/socketapi/socketapi.cpp [hunk 2]
```diff
@@ -1367,7 +1367,7 @@ void SocketApi::command_ASYNC_ASSERT_ICON_IS_EQUAL(const QSharedPointer<SocketAp
     if (value.name() == iconName) {
         job->resolve();
     } else {
-        job->reject("iconName " + iconName + " does not match: " + value.name());
+        job->reject(QStringLiteral("iconName ") + iconName + QStringLiteral(" does not match: ") + value.name());
```
Comment (TheOneRing) [inline src/gui/socketapi/socketapi.cpp:1370]: it could use .arg()

**Comment by TheOneRing** [comment]
Is the feature used anywhere?
It was intended for some kind of lets write our own squish approach?

**Comment by ownclouders** [comment]
Results for <strong>GUI-tests</strong> https://drone.owncloud.com/owncloud/client/13070/6/1

:boom: The GUI tests failed.

GUI Logs: https://cache.owncloud.com/public/owncloud/client/13070/guiReportUpload/index.html

<details>

Server Logs: https://cache.owncloud.com/public/owncloud/client/13070/guiReportUpload/serverlog.log

Screenshots:
- [share_file_and_folder_to_a_group.png](https://cache.owncloud.com/public/owncloud/client/13070/guiReportUpload/screenshots/share_file_and_folder_to_a_group.png)

</details>

File: src/gui/socketapi/socketapi.cpp (hunk 2 above)
Comment (erikjv) [reply src/gui/socketapi/socketapi.cpp:1370]: it *should* use `.arg()`. If it's a user-visible string, use `tr()`.

**Review by TheOneRing — APPROVED** [review]

**Pull request merged**